tests/components/samsungtv/test_config_flow.py

```diff
@@ -0,0 +1,388 @@
+        DOMAIN, context={"source": "user"}, data=MOCK_USER_DATA
+    )
+    assert result["type"] == "create_entry"
+    assert result["data"][CONF_MANUFACTURER] is None
+    assert result["data"][CONF_MODEL] is None
+    assert result["data"][CONF_ID] is None
+
+    # failed as already configured
+    result2 = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "ssdp"}, data=MOCK_SSDP_DATA
+    )
+    assert result2["type"] == "abort"
+    assert result2["reason"] == "already_configured"
+
+    # check updated device info
+    assert result["data"][CONF_MANUFACTURER] == "fake_manufacturer"
+    assert result["data"][CONF_MODEL] == "fake_model"
+    assert result["data"][CONF_ID] == "fake_uuid"
+
+
+async def test_autodetect_websocket(hass, remote):
+    """Test for send key with autodetection of protocol."""
+    with patch("homeassistant.components.samsungtv.config_flow.Remote") as remote:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "user"}, data=MOCK_USER_DATA
+        )
+        assert result["type"] == "create_entry"
+        assert result["data"][CONF_METHOD] == "websocket"
+        assert remote.call_count == 1
+        assert remote.call_args_list == [call(AUTODETECT_WEBSOCKET)]
+
+
+async def test_autodetect_auth_missing(hass, remote):
+    """Test for send key with autodetection of protocol."""
+    with patch(
+        "homeassistant.components.samsungtv.config_flow.Remote",
+        side_effect=[AccessDenied("Boom")],
+    ) as remote:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "user"}, data=MOCK_USER_DATA
+        )
+        assert result["type"] == "abort"
+        assert result["reason"] == "auth_missing"
+        assert remote.call_count == 1
+        assert remote.call_args_list == [call(AUTODETECT_WEBSOCKET)]
+
+
+async def test_autodetect_not_supported(hass, remote):
+    """Test for send key with autodetection of protocol."""
+    with patch(
+        "homeassistant.components.samsungtv.config_flow.Remote",
+        side_effect=[UnhandledResponse("Boom")],
+    ) as remote:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "user"}, data=MOCK_USER_DATA
+        )
+        assert result["type"] == "abort"
+        assert result["reason"] == "not_supported"
+        assert remote.call_count == 1
+        assert remote.call_args_list == [call(AUTODETECT_WEBSOCKET)]
+
+
+async def test_autodetect_legacy(hass, remote):
+    """Test for send key with autodetection of protocol."""
+    with patch(
+        "homeassistant.components.samsungtv.config_flow.Remote",
+        side_effect=[OSError("Boom"), mock.DEFAULT],
+    ) as remote:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "user"}, data=MOCK_USER_DATA
+        )
+        assert result["type"] == "create_entry"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4795320)